src/Extension/PublishableExtension.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Silverback API Component Bundle Project
+ *
+ * (c) Daniel West <daniel@silverback.is>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Silverback\ApiComponentBundle\Extension;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\ContextAwareQueryCollectionExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Silverback\ApiComponentBundle\Annotation\Publishable;
+use Silverback\ApiComponentBundle\Publishable\PublishableHelper;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @author Vincent Chalamon <vincent@les-tilleuls.coop>
+ */
+final class PublishableExtension implements QueryItemExtensionInterface, ContextAwareQueryCollectionExtensionInterface
+{
+    private PublishableHelper $publishableHelper;
+    private RequestStack $requestStack;
+    private ?Publishable $configuration;
+
+    public function __construct(PublishableHelper $publishableHelper, RequestStack $requestStack)
+    {
+        $this->publishableHelper = $publishableHelper;
+        $this->requestStack = $requestStack;
+    }
+
+    public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, array $identifiers, string $operationName = null, array $context = []): void
+    {
+        $configuration = $this->getConfiguration($resourceClass);
+        if (!$configuration || !($request = $this->requestStack->getCurrentRequest())) {
+            return;
+        }
+
+        if (!$this->isAllowed($context)) {
+            // User has no access to draft object
+            $this->updateQueryBuilderForUnauthorizedUsers($queryBuilder, $configuration);
+
+            return;
+        }
+
+        // Reset queryBuilder to prevent an invalid DQL
+        $queryBuilder->where('1 = 1');
+
+        $alias = $queryBuilder->getRootAliases()[0];
+        foreach ($identifiers as $identifier) {
+            // (o.id = :id AND o.publishedAt IS NOT NULL AND o.publishedAt <= :currentTime)
+            // OR ((o.publishedAt IS NULL OR o.publishedAt > :currentTime) AND o.publishedResource = :id)
```

Comment:
I don't think we need to check about the date anymore here because we know we have access either way and have granted access no matter what. We just want primarily a resource where publishedResource is the ID, followed by the resource ID if it doesn't exist.